cmake/OpenCVDetectOpenCL.cmake

```diff
@@ -2,14 +2,33 @@ set(OPENCL_FOUND ON CACHE BOOL "OpenCL library is found")
 if(APPLE)
   set(OPENCL_LIBRARY "-framework OpenCL" CACHE STRING "OpenCL library")
   set(OPENCL_INCLUDE_DIR "" CACHE PATH "OpenCL include directory")
-else(APPLE)
+else()
   set(OPENCL_LIBRARY "" CACHE STRING "OpenCL library")
   set(OPENCL_INCLUDE_DIR "${OpenCV_SOURCE_DIR}/3rdparty/include/opencl/1.2" CACHE PATH "OpenCL include directory")
   ocv_install_3rdparty_licenses(opencl-headers "${OpenCV_SOURCE_DIR}/3rdparty/include/opencl/LICENSE.txt")
-endif(APPLE)
+endif()
 mark_as_advanced(OPENCL_INCLUDE_DIR OPENCL_LIBRARY)
 
 if(OPENCL_FOUND)
+  if(WITH_OPENCL_D3D11_NV)
+    if(NOT EXISTS "${OPENCL_INCLUDE_DIR}/CL/cl_d3d11_ext.h")
+      message(STATUS "CL/cl_d3d11_ext.h not found in OPENCL_INCLUDE_DIR. Automatic detection has been turned on")
+      unset(OPENCL_INCLUDE_DIR CACHE)
+      find_path(OPENCL_INCLUDE_DIR
```

Comment:
We should not unset/rewrite `OPENCL_INCLUDE_DIR` from cache, we should disable the feature in case when the header was not found:
```.cmake
if(WITH_OPENCL_D3D11_NV AND EXISTS "${OPENCL_INCLUDE_DIR}/CL/cl_d3d11_ext.h")
  set(HAVE_OPENCL_D3D11_NV ON)
endif()
```
User wishing to enable this feature should check cmake output or turn on `ENABLE_CONFIG_VERIFICATION` cmake option to verify it automatically.

Reply:
Since `cl_d3d11_ext.h` has been moved to the default directory, the search will no longer be necessary.Done.